### PR TITLE
Xeno Rank Overhaul and Other Rank Things

### PR DESCRIPTION
**Pull request by Nakhi1**
I decided to go ahead and close some loopholes within the ranking system. Some subspecies weren't listed and thus had full access where the base species was restricted as one example. Genemodders, if they're to be human, are thus treated as human and removed from restriction. As well, the Marine Captain and Specialist Captain ranks are commented out for Chief of Security, Chief Medical Officer, and Chief Engineer. Annotated changes are below.

- Northern and Eastern Sergals are no longer allowed to be Marines. Base Sergals weren't allowed. Subspecies brought in line with base.
- Yeosa'Unathi no longer have access to all Marine ranks. Base Unathi and Yeteris'Unathi were restricted to Trooper ranks. Subspecies brought in line with base.
- Prometheans, due to being squishy and easily killed and destroyed, are civilian/contractor restricted pending future fixes to the species.
- TROOPER_ONLY rank restriction raised to E-4, allowing Unathi and Tajara to be Corporals instead of just Lance Corporals.
- Vasilissans allowed to be private security.
- Marine Captain and Specialist Captain O-3 alt ranks were removed. 'Captain' O-3 rank re-implemented for Chief Engineer, Chief of Security, and Chief Medical Officer.
- Genemods, if they're to be treated as humans with fur effectively, have been removed from a xenos whitelist. They have full access to all Marine ranks up to O-5.

### CODE REVIEW

**Comment by Roland410** [comment]
Still going to disagree with the promethean part of the PR, especially as the fixes are in the work and should be posted soon.

**Comment by Chaznoodles** [comment]
This is a solution to the inconsistency between species ranks, and I wholly support it.

**Comment by Karl-Franz-I** [comment]
This is a solution to the inconsistency between species ranks in SMC, and I wholly support it.

**Comment by MarinaGryphon** [comment]
> * Yeosa'Unathi are no longer allowed to be any marine rank.

Why, though.
I suggest separating out the bugfixes/oversights (sergals, genemodders) from the removals, given that those are going to be a lot more contentious.

**Comment by MarinaGryphon** [comment]
> * Genemods, if they're to be treated as humans with fur effectively,

Also, they're not. They can be any species. Maybe make it use the icon base instead, for genemodders.

**Comment by Nakhi1** [comment]
> Why, though.
> I suggest separating out the bugfixes/oversights (sergals, genemodders) from the removals, given that those are going to be a lot more contentious.
Read the full line, not the first sentence and look at the code. Yeosa were overlooked in assigning branches, allowing them to have access to any marine ranks. They have been brought in line with regular Unathi and the Veteris to be Troopers only.

> Also, they're not. They can be any species. Maybe make it use the icon base instead, for genemodders.
The lore on them in announcements or important documents says that they are treated generally as humans. And that's how most play them as.

File: maps/torch/torch_ranks_boh.dm [hunk 1]
```diff
@@ -63,15 +68,19 @@
...
+		/datum/species/sergal/eastern	= list(UNRESTRICTED, /datum/mil_branch/private_security),
 		/datum/species/akula		= list(UNRESTRICTED, /datum/mil_branch/private_security),
 		/datum/species/custom		= list(UNRESTRICTED, SEMIRESTRICTED),
 		/datum/species/humanathi	= list(UNRESTRICTED, SEMIRESTRICTED),
 		/datum/species/tajaran		= list(UNRESTRICTED, SEMIRESTRICTED),
-		/datum/species/vasilissan	= list(UNRESTRICTED),
+		/datum/species/vasilissan	= list(UNRESTRICTED, /datum/mil_branch/private_security),
 		/datum/species/vulpkanin	= list(UNRESTRICTED, SEMIRESTRICTED),
 		/datum/species/customhuman	= list(UNRESTRICTED, SEMIRESTRICTED),
+		/datum/species/shapeshifter/promethean	= list(/datum/mil_branch/civilian),
```
Comment (MarinaGryphon) [inline maps/torch/torch_ranks_boh.dm]:
```suggestion
```
Move this into a separate PR.

**Comment by tichys** [comment]
unmerge the pr,  race war incoming
